A26-grpclb-selection.md

```diff
@@ -0,0 +1,135 @@
+A26: gRPCLB Selection
+----
+* Author(s): Doug Fawley
+* Approver: a11r
+* Status: Draft
+* Implemented in: None
+* Last updated: 2019-10-22
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Modify the method for enabling gRPCLB to not require any special handling in
+core, allowing for stabilization of the gRPCLB protocol and load balancing
+policy.
+
+## Background
+
+The existing mechanism to enable and select gRPCLB is currently still
+experimental. In addition, gRPCLB itself is deprecated, to be replaced by xDS as
+the official advanced load balancing protocol for gRPC ([announcement
+email](https://groups.google.com/d/msg/grpc-io/0yGihF-EFQo/A4QKdXffBwAJ)). We
+would like to enable production uses of gRPCLB, but cannot do so if gRPCLB
+support is not stable. Therefore, we need a mechanism for selecting gRPCLB that
+can be declared stable, yet not create long-lasting maintenance problems.
+
+## Proposal
+
+C core will promote its existing channel arg that configures the DNS resolver to
+fetch SRV records to "stable".
+
+In Go and Java, the default DNS resolver will not fetch SRV records. However,
+when gRPCLB is imported, it will override the default DNS resolver with a
+resolver that does fetch SRV records. Justification for this approach: without
+the gRPCLB LB policy, balancer addresses have no value. Users that do not wish
+to use gRPCLB will not take the performance penalty caused by attempting to
+resolve the balancer addresses through SRV records.
+
+Additionally, in all languages, to remove special handling of gRPCLB from the
+channel:
+- The loadBalancingConfig field in service config will be the only way to
+  specify the "grpclb" policy.
+- Balancer addresses will now only be provided via side-channel data passed from
+  the resolver to the balancer (similar to the xDS client instance).
```

Comment:
It's not called out -- are the balancer addresses now included in the loadBalancingConfig?

Reply:
No. They are through a side-channel API that we don't have a name for. In Java it is via the arbitrary Attributes passed from NameResolver (not in EAGs).